iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -659,19 +651,43 @@ popo::InterfacePortData* PortManager::acquireInterfacePortData(capro::Interfaces
     }
 }
 
+void PortManager::publishCurrentServiceRegistry() const noexcept
+{
+    // Send the new serviceRegistry here and ring the bell to inform all ServiceDiscovery instances about the change
+    if (m_serviceRegistryPublisherPortData)
+    {
+        PublisherPortUserType publisher(m_serviceRegistryPublisherPortData);
```

Comment:
Can the `Portmanager` have this `PublisherPortUserType publisher` as a member? I know the object creation is lightweight but I am just wondering why we need to create it each time we send.

Reply:
I think it would be possible in theory. However, we are using the pattern like above everywhere in the code when working on shared memory ports data. The object changing the data in shared memory lives on the stack and is created on demand. Hence, I would like to stick with the current approach.

Reply:
Yeah it is ok but slightly suboptimal.